File: APEC.RRHH.api/Core/Models/Job.cs
using Core.Enums;
using System;
using System.Collections.Generic;
using Core.Contracts;

namespace Core.Models
{
    public sealed class Job : IEntityBase
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public RiskLevel RiskLevel { get; set; }
        public int MinimumSalary { get; set; }
        public int MaximumSalary { get; set; }
        public int QuantityOfEmployeesNeeded { get; set; }
        public FeatureStatus Status { get; set; }
        public CurrencyType CurrencyType { get; set; }
        public Departament Departament { get; set; }
        public Guid DepartamentId { get; set; }
        public ISet<Employee> Employees { get; set; }
        public ISet<JobCompetence> JobCompetences { get; set; }
        public ISet<JobLanguage> JobLanguages { get; set; }
        public int MinimumYearsExperienceLaboral { get; set; }
    }
}

File: APEC.RRHH.webapp/src/app/models/job.ts
import { CurrencyType } from '../enums/currency-type.enum';
import { FeatureStatus } from '../enums/feature-status';
import { RiskLevel } from '../enums/risk-level.enum';
import { Employee } from './employee';
import { Language } from './language';
import { Competence } from './competence';
import { CandidateInterview } from './candidate-interview';

export class Job {
    id: string;
    name: string;
    minimumSalary: number;
    maxSalary: number;
    minimumYearsExperienceLaboral: number;
    quantityOfEmployeesNeeded: number;
    departamentId: string;
    currencyType: CurrencyType;
    status: FeatureStatus;
    riskLevel: RiskLevel;
    employees: Employee[];
    jobCompetences: Competence[];
    jobLanguages: Language[];
    interviews: CandidateInterview[];
}
